Fixes #1967: Correctly handle mocks with limited life-cycle in listeners.

raphw opened the pull request:
Due to the limited lifetime of static mocks, they cannot be validated within a Mockito session since they might have expired when the validation is applied. This patch excludes static mocks from all validation. To support static mocks in the regular session validation, we would rather need to trigger the validation upon release of any static mock.

raphw commented:
It only tests part of the issue. For actually testing it, we would need to run the JUnit tests with the inline mock maker and create subprojects for it. I am going on a two week vacation tomorrow, so I did not want to open that barrel of wine tonight but I validated the working manually. If you have some time, could you add additional subprojects? I validated the runners manually, however.

TimvdLippe commented:
Ah okay. I got confused as the commit mentions that the issue is fixed, so that's why I thought we had test the whole scenario. Please go on vacation, we can figure it out later 😄

raphw commented:
I added an additional project for the JUnit Jupiter extension and added a JUnit rule for the inline mock maker tests. I hope this should avoid any future regressions.
